plugin.video.popcorntimes/main.py

```diff
@@ -0,0 +1,191 @@
+	listItem.setArt({'thumb':ICON,'fanart':FANART})
+	addDirectoryItem(_handle, get_url(action='listing', url="/de/top-filme"), listItem, True)
+	listItem = ListItem(LANGUAGE(30005))
+	listItem.setProperty('IsPlayable', 'false')
+	listItem.setArt({'thumb':ICON,'fanart':FANART})
+	addDirectoryItem(_handle, get_url(action='listing', url="/de/neu"), listItem, True)
+	listItem = ListItem(LANGUAGE(30006))
+	listItem.setProperty('IsPlayable', 'false')
+	listItem.setArt({'thumb':ICON,'fanart':FANART})
+	addDirectoryItem(_handle, get_url(action='genre'), listItem, True)
+	endOfDirectory(_handle)
+
+
+def list_genre():
+	req = s.get(base_url + "/de/genres")
+	soup = BeautifulSoup(req.text, "html.parser")
+	genres_h3 = soup.find("div", class_ = "pt-bordered-tiles").find_all("div", class_ = "pt-movie-tile")
+	for h3 in genres_h3:
+		link = h3.find("a")
+		img = h3.find("img")
+		listItem = ListItem(img['alt'])
+		listItem.setProperty('IsPlayable', 'false')
+		img = img['data-src']
+		if img is None:
+			img = ICON
+		else:
+			img = "https:" + img
+
+		listItem.setArt({'thumb':ICON,'fanart':FANART,'poster':img})
+		addDirectoryItem(_handle, get_url(action='listing', url=link.get("href")), listItem, True)
+
+	endOfDirectory(_handle)
+
+
+def list_movies(url):
+	xbmcplugin.setContent(_handle, 'Movies')
+	req = s.get(base_url + url)
+	soup = BeautifulSoup(req.text, "html.parser")
+	mov_divs = soup.find_all("div", class_ = "pt-movie-tile-full")
+	for div in mov_divs:
+		if div.find("a") is not None:
+			img = div.find("a").find("img")
+			title = img["alt"]
+
+			if div.find("p", class_ = "pt-tile-desc") is not None:
+				plot = div.find("p", class_ = "pt-tile-desc").text
+			if div.find("p", attrs={'class': None}) is not None:
+				plot = div.find("p", attrs={'class': None}).text
+
+			pt_time = div.find("p", class_ = "pt-video-time")
+			if pt_time is not None:
+				year = pt_time.text.split('|',1)[0].strip()
+				duration = pt_time.find("span").text.split(' ',1)[0].strip()
+			try:
+				year = int(year)
+			except:
+				year = None
+			try:
+				duration = int(duration) * 60
+			except:
+				duration = None
+			img = div.find("a").find("img").get("data-src")
+			if img is None:
+				img = div.find("a").find("img").get("src")
+
+			listItem = ListItem(title)
+			listItem.setProperty('IsPlayable', 'true')
+			listItem.setArt({ 'fanart':FANART,'poster': "https:" + img }),
+			listItem.setInfo('video', {'title': title, 'plot': plot, 'year': year, 'duration': duration })
+			mov_url = base_url + div.find("a").get("href")
+			addDirectoryItem(_handle, get_url(action='play', url=mov_url), listItem, False)
+
+	endOfDirectory(_handle)
+	xbmcplugin.setContent(_handle, 'Movies')
+
+
+def play(movie_url):
+	title = movie_url
+	liz = xbmcgui.ListItem( '')
+	streamUrl = getStream(movie_url)
+	if streamUrl: 
+		fullurl = streamUrl.strip() + "|Referer=" + movie_url.strip()
+		liz.setPath(fullurl)
+		xbmcplugin.setResolvedUrl(_handle, True, liz)
+	#else:
```

Comment:
you might want to error out here, with a xbmcgui.Dialog().ok for example - not mandatory

Reply:
Good point, will keep it in mind for a later version.